packages/data/src/factory.ts

```diff
@@ -60,12 +73,10 @@ export function createRegistrySelector( registrySelector ) {
 	 * Flag indicating that the selector is a registry selector that needs the correct registry
 	 * reference to be assigned to `selector.registry` to make it work correctly.
 	 * be mapped as a registry selector.
-	 *
-	 * @type {boolean}
 	 */
 	wrappedSelector.isRegistrySelector = true;
 
-	return wrappedSelector;
+	return wrappedSelector as unknown as Selector;
```

Comment:
Hmm, does this actually make sense? Aren't we augmenting `Selector` with a few additional properties (`isRegistrySelector`, `registry`, `__unstableNormalizeArgs`)?

Reply:
TypeScript complains

```
Type 'WrappedSelector' is not assignable to type 'Selector'.
 
'WrappedSelector' is assignable to the constraint of type 'Selector', but 'Selector' could be instantiated with a different subtype of constraint '(...args: any[]) => any'.ts(2322)
```

<img width="1170" alt="image" src="https://github.com/user-attachments/assets/c865974b-e541-4ca4-b1d6-0c5f34f0e1d8">

Reply:
I have extracted the type and fixed the assertion. Thanks